src/parser.js

```diff
@@ -1148,6 +1156,19 @@ function mapSwitchStatement(node, meta) {
 }
 
 function mapForStatement(node, meta) {
+  let blockStatement = mapBlockStatement(node.body, meta);
+
+  if (node.guard) {
+    const guardBase = node.guard.expression || node.guard;
+
+    blockStatement = b.blockStatement([
+      b.ifStatement(
+        mapOp(guardBase, meta),
```

Comment:
I *think* this is covered by the [it handles existential logical expression](https://github.com/Shopify/decaf/pull/7/files#diff-c1129c8b045390789fa8ff62f2c6b4a9R2682) test.  

The `foo for foo in foos` thing doesn't compile to realistic JS, so I may be misunderstanding, or there may be a typo in your example?